### PR TITLE
[TensorExpr] Add lowerings for scalar binary ops (+,-,*,/,&,|,^,<<,>>,cmp).

### PR DESCRIPTION
**Pull request by ZolotukhinM**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #71298
* #71186

Differential Revision: [D33576534](https://our.internmc.facebook.com/intern/diff/D33576534)

### CODE REVIEW

**Comment by ZolotukhinM** [comment]
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33576534).

**Comment by ZolotukhinM** [comment]
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33576534).

**Comment by ZolotukhinM** [comment]
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33576534).

**Comment by ZolotukhinM** [comment]
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33576534).

File: torch/csrc/jit/tensorexpr/lowerings.cpp [hunk 1]
```diff
@@ -80,22 +80,114 @@ int nnc_lowerings_lazy_registration() {
...
+      });
+  DEFINE_COMPARISON_SCALAR_OP_LOWERING(lt, cast<bool>(a < b))
+  DEFINE_COMPARISON_SCALAR_OP_LOWERING(le, cast<bool>(a <= b))
+  DEFINE_COMPARISON_SCALAR_OP_LOWERING(eq, cast<bool>(a == b))
+  DEFINE_COMPARISON_SCALAR_OP_LOWERING(ne, cast<bool>(a != b))
+  DEFINE_COMPARISON_SCALAR_OP_LOWERING(gt, cast<bool>(a > b))
+  DEFINE_COMPARISON_SCALAR_OP_LOWERING(ge, cast<bool>(a >= b))
+#undef DEFINE_COMPARISON_SCALAR_OP_LOWERING
+
+#define DEFINE_INT_SCALAR_OP_LOWERING(op_name, op)                        \
```
Comment (navahgar) [inline torch/csrc/jit/tensorexpr/lowerings.cpp:151]: nit: s/DEFINE_INT_SCALAR_OP_LOWERING/DEFINE_BITWISE_SCALAR_OP_LOWERING

File: torch/csrc/jit/tensorexpr/lowerings.cpp [hunk 2]
```diff
@@ -80,22 +80,114 @@ int nnc_lowerings_lazy_registration() {
...
+            [](const ExprHandle& a, const ExprHandle& b) { return op; }); \
+      });
+  DEFINE_INT_SCALAR_OP_LOWERING(__and__, boolToInteger(a) & boolToInteger(b))
+  DEFINE_INT_SCALAR_OP_LOWERING(__or__, boolToInteger(a) | boolToInteger(b))
+  DEFINE_INT_SCALAR_OP_LOWERING(__xor__, boolToInteger(a) ^ boolToInteger(b))
+  DEFINE_INT_SCALAR_OP_LOWERING(__lshift__, a << b)
+  DEFINE_INT_SCALAR_OP_LOWERING(__rshift__, a >> b)
+#undef DEFINE_INT_SCALAR_OP_LOWERING
+
+#define DEFINE_BOOL_SCALAR_OP_LOWERING(op_name, op)                       \
```
Comment (navahgar) [inline torch/csrc/jit/tensorexpr/lowerings.cpp:172]: nit: s/DEFINE_BOOL_SCALAR_OP_LOWERING/DEFINE_LOGICAL_SCALAR_OP_LOWERING

File: test/test_jit_fuser_te.py [hunk 3]
```diff
@@ -1386,6 +1386,62 @@ def apply(fn):
...
+          return (%z)"""
+
+        binary_ops = [
+                "aten::mul",
+                "aten::add",
+                "aten::sub",
+                "aten::div",
+                "aten::lt",
+                "aten::le",
+                "aten::eq",
```
Comment (navahgar) [inline test/test_jit_fuser_te.py:1404]: `aten::ne` missing

**Review by navahgar — APPROVED** [review]
LGTM.

**Comment by ZolotukhinM** [comment]
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33576534).

**Comment by ZolotukhinM** [comment]
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33576534).

**Comment by ZolotukhinM** [comment]
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33576534).

**Pull request closed without merging**